src/modules/builtin/touch.rs

```diff
@@ -0,0 +1,65 @@
+use crate::fragments;
+use crate::modules::expression::expr::Expr;
+use heraclitus_compiler::prelude::*;
+use crate::modules::prelude::*;
+use crate::modules::types::{Type, Typed};
+
+#[derive(Debug, Clone)]
+pub struct Touch {
+    value: Expr,
+}
+
+impl SyntaxModule<ParserMetadata> for Touch {
+    syntax_name!("TouchFile");
+
+    fn new() -> Self {
+        Touch { value: Expr::new() }
+    }
+
+    fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
+        let position = meta.get_index();
+        token(meta, "touch")?;
+
+        if token(meta, "(").is_ok() {
+            syntax(meta, &mut *self.value)?;
```

Comment:
@Mte90 consider not dereferencing expression here:
```suggestion
            syntax(meta, &mut self.value)?;
```